app/utils/file.js

```diff
@@ -29,11 +31,79 @@ function decrypt(file, key) {
   return file - key;
 }
 
-function shred(file) {
-  // erasure coding with reed solomon
-  const shreds = file;
+// inputFile is the path-name of the file to be shredded
+// Parity is a multiple of the number of shreds in the original file
+// The % of shreds we can lose is = (Parity/(Parity+1))*100
+function shredFile(parity, shredLength, inputFile) {
+  const bitmap = fs.readFileSync(inputFile);
+  const dataBuffer = new Buffer(bitmap);
+  const shardLength = shredLength;
+  const dataShards = Math.ceil(dataBuffer.length / shardLength);
+  const parityShards = parity * dataShards;
+  const totalShards = dataShards + parityShards;
+  // Create the parity buffer
+  const parityBuffer = Buffer.alloc(parityShards * shardLength);
+  const bufferOffset = 0;
+  const bufferSize = shardLength * totalShards;
+  const shardOffset = 0;
+  const shardSize = shardLength - shardOffset;
 
-  return shreds;
+  const buffer = Buffer.concat([
+    dataBuffer,
+    parityBuffer
+  ], bufferSize);
+
+  const rs = new ReedSolomon(dataShards, parityShards);
+  rs.encode(
+    buffer,
+    bufferOffset,
+    bufferSize,
+    shardLength,
+    shardOffset,
+    shardSize,
+    (error) => {
+      if (error) throw error;
+      // Parity shards now contain parity data.
+    }
+  );
+  // writing data shards as files
+  for (let i = 0; i < totalShards; i += i) {
+    // Generate shred IDs to name the shreds
+    fs.writeFileSync(`${i}_${Math.random()}`, buffer.slice(i * shardLength, (i + 1) * shardLength));
+  }
+}
+
+// shredsBuffer: a Buffer containing the shreds to be recovered,
+// targets: a variable containing the indecies of the missing shreds
+// dataShreds: Math.ceil(dataBuffer.length / shardLength)
+// recoverdFile: the name of the file to be recovered
+function recoverFile(shredsBuffer, targets, parity, shredLength, dataShreds, recoveredFile) {
+  const buffer = new Buffer(shredsBuffer);
```

Comment:
here you are reading shredsbuffer as a  buffer, that means you will have  a method that reads files(shreds) and concatenate them to a big buffer and call method recoverFile(..) , right ?